temporalio/bridge/src/runtime.rs

```diff
@@ -118,12 +118,31 @@ pub fn init_runtime(telemetry_config: TelemetryConfig) -> PyResult<RuntimeRef> {
         }
     }
 
+    let task_locals = Python::with_gil(|py| {
+        // Event loop is assumed to be running at this point
+        let locals = pyo3_asyncio::TaskLocals::with_running_loop(py)?.copy_context(py)?;
+        PyResult::Ok(locals)
+    })?;
+
     // Create core runtime which starts tokio multi-thread runtime
     let mut core = CoreRuntime::new(
         telemetry_build
             .build()
             .map_err(|err| PyValueError::new_err(format!("Invalid telemetry config: {}", err)))?,
-        tokio::runtime::Builder::new_multi_thread(),
+        TokioRuntimeBuilder {
+            inner: tokio::runtime::Builder::new_multi_thread(),
+            lang_on_thread_start: Some(move || {
```

Comment:
Can you help me understand this a bit? Looking at https://github.com/awestlake87/pyo3-asyncio/blob/master/src/tokio.rs I don't see where they are building a Tokio runtime w/ custom thread starter to add task locals for every thread. I wonder if we can just make/set the task locals with the event loop inside the one place they are needed later? The current event loop is available there too. I also see they use https://docs.rs/tokio/latest/tokio/macro.task_local.html, unsure if we could too.

Reply:
> What one place are you thinking of?

Ah, yes, my mistake. So question, can any of these Python references information be captured and _moved_ into the `CustomSlotSupplierOfType` or as the result of `with_gil`? I admit I haven't dug into this, but I know I found in Ruby a qay to keep "opaque" references that needed the GIL to use them, but were still Send/Sync so that I could move around (and they were prevented from being collected by Python while still having a reference in Rust).

If that's possible, I think we should put the `TaskLocals` on the `CustomSlotSupplierOfType`.

Reply:
Done. Honestly not sure it's any better, really, but done regardless

Reply:
True, though it is nice that it's worker specific and not actually setting any task locals